### PR TITLE
Remove remainders of legacy setting 'allow_experimental_query_cache'

### PR DESCRIPTION
**Pull request by rschu1ze**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

**Comment by robot-clickhouse-ci-2** [comment]
This is an automated comment for commit 07147ef88805ca09b5cd7120ecf60d71c1d94d55 with description of existing statuses. It's updated for the latest CI running
The full report is available [here](https://s3.amazonaws.com/clickhouse-test-reports/52685/07147ef88805ca09b5cd7120ecf60d71c1d94d55/ci_running.html)
The overall status of the commit is 🟡 pending

<table><thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody><tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>🟡 pending</td></tr>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>🟢 success</td></tr>
</table>

**Comment by nickitat** [comment]
what is the policy to remove settings? I'm afraid that we will cause problems for people that use this setting now (queries will fail after upgrade) without getting much benefit.

**Comment by rschu1ze** [comment]
Don't know about the general policy (probably a year or so). In this case the setting was experimental - IMHO there is no need to support it for another year (it should not have been enabled in productive deployments anyways).

**Review by nickitat — APPROVED** [review]

**Comment by nickitat** [comment]
> Don't know about the general policy (probably a year or so). In this case the setting was experimental - IMHO there is no need to support it for another year (it should not have been enabled in productive deployments anyways).

ok

**Review by alexey-milovidov — APPROVED** [review]

File: src/Core/Settings.h [hunk 1]
```diff
@@ -808,7 +808,6 @@ class IColumn;
     MAKE_OBSOLETE(M, UInt64, merge_tree_clear_old_parts_interval_seconds, 1) \
     MAKE_OBSOLETE(M, UInt64, partial_merge_join_optimizations, 0) \
     MAKE_OBSOLETE(M, MaxThreads, max_alter_threads, 0) \
-    MAKE_OBSOLETE(M, Bool, allow_experimental_query_cache, true) \
```
Comment (nikitamikhaylov) [inline src/Core/Settings.h:811]: This change is backward incompatible...

**Comment by tavplubix** [comment]
> what is the policy to remove settings?

We remove obsolete settings after one year or 2 LTS releases (not earlier).

> I'm afraid that we will cause problems for people that use this setting now

Yes, it caused some problems for us :) 

> In this case the setting was experimental - IMHO there is no need to support it for another year 

We don't need to do anything to support it. Just keep it in the list of obsolete settings, it's one line. We can even keep obsolete settings forever in that list, it's not a big deal

> (it should not have been enabled in productive deployments anyways).

Well, it could be disabled: `<allow_experimental_query_cache>0</allow_experimental_query_cache>`

**Pull request merged**